src/common.rs

```diff
@@ -397,6 +397,52 @@ impl PruneOptions {
     }
 }
 
+/// Options for [crate::sync::mount]
+///
+/// Mount an archive or repository as a FUSE filesystem. This is useful for
+/// browsing archives or repositories and interactive restoration.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct MountOptions {
+    /// Path to the repository or archive
+    ///
+    /// Example values:
+    /// - `/tmp/foo`
+    /// - `/tmp/foo::my-archive`
+    /// - `user@example.com:/opt/repo`
+    /// - `user@example.com:/opt/repo::archive`
+    /// - `ssh://user@example.com:2323:/opt/repo`
+    pub repository_or_archive: String,
```

Comment:
What do you think about splitting into `MountRepositoryOptions` and `MountArchiveOptions`. For repositories, I'd like to support the `--last N` and `--first N` options as well as the `--glob-archives GLOB` option. Maybe also the sorting option.

As an alternative, we could make `MountOptions` an enum with an archvive and a repository variant.

I lean to the latter option

Reply:
Hmm good idea. I wonder if `repository_or_archive` could actually hold the enum variant as the rest of the options should be the same. I'll see how that looks

Reply:
(we discussed this in [another thread](https://github.com/myOmikron/borgbackup/pull/4#discussion_r1179805254), so I'll resolve this)